### ISSUE TITLE
NotImplementedError from statically partitioned asset dependent on all partitions of statically partitioned asset

### ISSUE THREAD

**Issue by craustin** [dagster-io/dagster#19179]
### Dagster version

1.5.13

### What's the issue?

When I right-click-materialize `repro_asset_1` (all three partitions), I get the below `NotImplementedError`:

```python
from dagster import AllPartitionMapping
from dagster import asset
from dagster import AssetDep
from dagster import StaticPartitionsDefinition


@asset(
    partitions_def=StaticPartitionsDefinition(['A', 'B', 'C']),
)
def repro_asset_1():
    pass


@asset(
    partitions_def=StaticPartitionsDefinition(['D', 'E', 'F']),
    deps=[AssetDep(repro_asset_1, partition_mapping=AllPartitionMapping())],
)
def repro_asset_2():
    pass
```

```
NotImplementedError
  File "/home/caustin/envs/ecm-macro/lib/python3.10/site-packages/dagster_graphql/implementation/utils.py", line 125, in _fn
    return fn(*args, **kwargs)
  File "/home/caustin/envs/ecm-macro/lib/python3.10/site-packages/dagster_graphql/implementation/utils.py", line 56, in _fn
    result = fn(self, graphene_info, *args, **kwargs)
  File "/home/caustin/envs/ecm-macro/lib/python3.10/site-packages/dagster_graphql/schema/roots/mutation.py", line 318, in mutate
    return create_and_launch_partition_backfill(graphene_info, backfillParams)
  File "/home/caustin/envs/ecm-macro/lib/python3.10/site-packages/dagster_graphql/implementation/execution/backfill.py", line 243, in create_and_launch_partition_backfill
    backfill = PartitionBackfill.from_asset_partitions(
  File "/home/caustin/envs/ecm-macro/lib/python3.10/site-packages/dagster/_core/execution/backfill.py", line 379, in from_asset_partitions
    asset_backfill_data = AssetBackfillData.from_asset_partitions(
  File "/home/caustin/envs/ecm-macro/lib/python3.10/site-packages/dagster/_core/execution/asset_backfill.py", line 554, in from_asset_partitions
    target_subset |= asset_graph.bfs_filter_subsets(
  File "/home/caustin/envs/ecm-macro/lib/python3.10/site-packages/dagster/_core/definitions/asset_graph.py", line 620, in bfs_filter_subsets
    partition_mapping.get_downstream_partitions_for_partitions(
  File "/home/caustin/envs/ecm-macro/lib/python3.10/site-packages/dagster/_core/definitions/partition_mapping.py", line 204, in get_downstream_partitions_for_partitions
    raise NotImplementedError()
```

### What did you expect to happen?

I expected all three partitions of `repro_asset_1` to be materialized in a backfill.

### How to reproduce?

See above

### Deployment type

Dagster Helm chart

### Message from the maintainers

Impacted by this issue? Give it a 👍! We factor engagement into prioritization.

**Comment by sryza** [comment]
Thanks for reporting this @craustin . @clairelin135 - thoughts on how to fix this? I think we assumed that `AllPartitionMapping` would be used when the downstream asset is unpartitioned, but I do think this is a reasonable usage, and in this case `AllPartitionMapping. get_downstream_partitions_for_partitions` should just be returning all of the partitions in the downstream asset?

**Comment by clairelin135** [comment]
> in this case AllPartitionMapping. get_downstream_partitions_for_partitions should just be returning all of the partitions in the downstream asset

Yup, agree that we should just implement the `AllPartitionMapping. get_downstream_partitions_for_partitions` method. I'll file a fix for this this week

**Comment by craustin** [comment]
@clairelin135 @gibsondan I proposed PR https://github.com/dagster-io/dagster/pull/19375 to address this

**Comment by clairelin135** [comment]
@craustin thanks, just responded to you on the PR

**Comment by abhischekt** [comment]
Hi @clairelin135 this wasn't included in the `1.6.2` release right? Do you know which version this fix will be released with?

**Comment by craustin** [comment]
@abhischekt I believe the fix was included in `1.6.2`. I see it in the commits included in that release. I haven't verified w/ that build yet, though.

**Comment by abhischekt** [comment]
> @abhischekt I believe the fix was included in `1.6.2`. I see it in the commits included in that release. I haven't verified w/ that build yet, though.

I don't see the change in the `1.6.2` tag here: https://github.com/dagster-io/dagster/blob/1bc63fdfd4efe3903342ef0c7c6b651727ea41a9/python_modules/dagster/dagster/_core/definitions/partition_mapping.py#L196-L208

**Comment by craustin** [comment]
Ah sorry- you are right ... it wasn't included.

**Comment by yuhan** [comment]
Hi! It will be included in tomorrow's 1.6.3 release.

**Comment by abhischekt** [comment]
> Hi! It will be included in tomorrow's 1.6.3 release.

Wonderful news! Thank you so much for confirming!
